CommonUI/Sources/CommonUI/MovieUI/MovieDetails/MovieDetailsViewModel.swift

```diff
@@ -0,0 +1,25 @@
+import SwiftUI
+import UserData
+import Combine
+import ApiService
+
+public class MovieDetailsViewModel: ObservableObject {
+    private let userData = UserData.shared
+    @Published var isFavorite = false
+    @Published var movie: Movie
+    
+    private var cancellables: Set<AnyCancellable> = []
+    
+    init(movie: Movie) {
+        self.movie = movie
+        self.fetchFavorite()
```

Comment:
remove this. Since the `MovieDetailsView` calls `fetchFavorite()` during `.onAppear`. 

Having `fetchFavorite` here causes it to be called twice

Reply:
removed it.